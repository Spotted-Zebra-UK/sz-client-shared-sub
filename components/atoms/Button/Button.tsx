import React, { FunctionComponent } from 'react';

export interface IButton {
  onClick?: (id?: string) => void;
  id?: string;
  className?: string;
  isDisabled?: boolean;
}

const Button: FunctionComponent<IButton> = ({
  children,
  onClick,
  id,
  className,
  isDisabled,
}) => {
  const handleClick = () => {
    if (onClick) {
      if (id) {
        onClick(id);
      } else {
        onClick();
      }
    }
  };

  return (
    <button
      className={`Button${className ? ` ${className}` : ''}`}
      type="button"
      onClick={handleClick}
      id={id}
      disabled={isDisabled}
    >
      {children}
    </button>
  );
};

export default Button;
