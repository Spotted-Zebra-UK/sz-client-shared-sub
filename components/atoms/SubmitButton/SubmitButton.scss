@import '../../../variables';

.SubmitButton {
  background-color: $green;
  border: 0;
  border-radius: $button-border-radius;
  color: $white;
  cursor: pointer;
  font-size: $p1-font-size;
  font-weight: $medium-bold;
  line-height: $medium-line-height;
  height: 56px;
  width: 251px;
}
