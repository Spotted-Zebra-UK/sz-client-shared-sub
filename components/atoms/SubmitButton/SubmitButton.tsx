import React, { FunctionComponent } from 'react';

interface ISubmitButton {
  className?: string;
  isDisabled?: boolean;
}

const SubmitButton: FunctionComponent<ISubmitButton> = ({
  children,
  className,
  isDisabled = false,
}) => {
  return (
    <button
      className={`SubmitButton${className ? ` ${className}` : ''}`}
      type="submit"
      disabled={isDisabled}
    >
      {children}
    </button>
  );
};

export default SubmitButton;
