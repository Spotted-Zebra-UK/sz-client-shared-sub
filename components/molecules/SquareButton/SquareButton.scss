@import '../../../variables';

.SquareButton {
  border: 0;
  border-radius: $button-border-radius;
  cursor: pointer;
  font-size: $p1-font-size;
  font-weight: $medium-bold;
  line-height: $medium-line-height;
  height: 56px;
  width: 251px;
}

.SquareButton:active {
  transform: translateY(-2px);
  box-shadow: 0 5px 10px rgba(0, 0, 0, 0.2);
}

.SquareButton--Green {
  background-color: $green;
  color: $white;
}

.SquareButton--Purple {
  background-color: $purple;
  color: $white;
}

.SquareButton--White {
  background-color: $white;
  color: $black;
}

.SquareButton--Disabled {
  background-color: rgba($color: $black, $alpha: 0.3);
  cursor: not-allowed;
}
