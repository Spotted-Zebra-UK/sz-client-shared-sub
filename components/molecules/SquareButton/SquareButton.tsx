import './SquareButton.scss';
import React, { FC } from 'react';
import { TButtonType } from '../../../interfaces/button';
import { TColor } from '../../../interfaces/colors';
import Button from '../../atoms/Button/Button';
import SubmitButton from '../../atoms/SubmitButton/SubmitButton';

export interface ISquareButton {
  onClick?: (id?: string) => void;
  id?: string;
  className?: string;
  color?: TColor;
  isDisabled?: boolean;
  type?: TButtonType;
}

const SquareButton: FC<ISquareButton> = props => {
  const { color, type, children, className, ...restProps } = props;

  const parsedClassName = `SquareButton${
    className ? ` ${className}` : ''
  } SquareButton--${color || 'Green'}`;

  if (type === 'submit') {
    return (
      <SubmitButton className={parsedClassName} {...restProps}>
        {children}
      </SubmitButton>
    );
  }

  return (
    <Button className={parsedClassName} {...restProps}>
      {children}
    </Button>
  );
};

export default SquareButton;
