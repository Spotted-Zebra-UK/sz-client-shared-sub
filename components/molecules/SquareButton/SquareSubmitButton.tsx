import './SquareButton.scss';
import { FC } from 'react';
import { TColor } from '../../../interfaces/colors';

export interface ISquareButton {
  onClick?: (
    event: React.MouseEvent<HTMLButtonElement, MouseEvent> | undefined
  ) => void;
  id?: string;
  className?: string;
  color?: TColor;
  isDisabled?: boolean;
}

const SquareSubmitButton: FC<ISquareButton> = props => {
  const { color, children, className, onClick, ...restProps } = props;

  const parsedClassName = `SquareButton${
    className ? ` ${className}` : ''
  } SquareButton--${color || 'Green'}${
    restProps.isDisabled ? ' SquareButton--Disabled' : ''
  }`;

  return (
    <button
      onClick={onClick}
      type="button"
      className={parsedClassName}
      {...restProps}
    >
      {children}
    </button>
  );
};

export default SquareSubmitButton;
