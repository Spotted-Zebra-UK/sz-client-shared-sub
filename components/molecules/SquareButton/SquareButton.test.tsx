import { mount, shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import React from 'react';
import Button from '../../atoms/Button/Button';
import SubmitButton from '../../atoms/SubmitButton/SubmitButton';
import SquareButton from './SquareButton';

describe('SquareButton', () => {
  let onClick: (id?: string) => void;

  beforeEach(() => {
    onClick = jest.fn();
  });

  describe('Submit', () => {
    it('should render correctly when type is submit', () => {
      const wrapper = shallow(<SquareButton type="submit" />);

      expect(toJson(wrapper)).toMatchSnapshot();
      expect(wrapper.find(SubmitButton)).toHaveLength(1);
      expect(wrapper.hasClass('SquareButton'));
      expect(wrapper.hasClass('SquareButton--Green'));
    });
  });

  describe('Button', () => {
    it('should render correctly when type not provided', () => {
      const wrapper = shallow(<SquareButton onClick={onClick} />);

      expect(toJson(wrapper)).toMatchSnapshot();
      expect(wrapper.find(Button)).toHaveLength(1);
      expect(wrapper.hasClass('SquareButton'));
      expect(wrapper.hasClass('SquareButton--Green'));
    });

    it('should render correctly when type is button', () => {
      const wrapper = shallow(<SquareButton type="button" onClick={onClick} />);

      expect(toJson(wrapper)).toMatchSnapshot();
      expect(wrapper.find(Button)).toHaveLength(1);
    });

    it('should render correctly when color, isDisabled and className provided', () => {
      const className = 'test-className';
      const wrapper = shallow(
        <SquareButton
          color="Purple"
          isDisabled
          className={className}
          onClick={onClick}
        />
      );

      expect(wrapper.hasClass('SquareButton--Purple')).toBe(true);
      expect(wrapper.hasClass(className)).toBe(true);
    });

    it('should call onClick on button click', () => {
      const wrapper = mount(<SquareButton onClick={onClick} />);

      wrapper.find('button').simulate('click');

      expect(onClick).toBeCalled();
    });
  });
});
