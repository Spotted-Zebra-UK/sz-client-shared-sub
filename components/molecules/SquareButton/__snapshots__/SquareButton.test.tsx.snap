// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SquareButton Button should render correctly when type is button 1`] = `
<Button
  className="SquareButton SquareButton--Green"
  onClick={[MockFunction]}
/>
`;

exports[`SquareButton Button should render correctly when type not provided 1`] = `
<Button
  className="SquareButton SquareButton--Green"
  onClick={[MockFunction]}
/>
`;

exports[`SquareButton Submit should render correctly when type is submit 1`] = `
<SubmitButton
  className="SquareButton SquareButton--Green"
/>
`;
